### PR TITLE
Bump puma from 3.9.1 to 3.11.3

### PR DESCRIPTION
**Pull request by greysteil**
Bumps [puma](https://github.com/puma/puma) from 3.9.1 to 3.11.3.
<details>
<summary>Changelog</summary>

*Sourced from puma's [changelog](https://github.com/puma/puma/blob/master/History.md).*

> ## 3.11.3 / 2018-03-05
> 
> * 3 bugfixes:
>   * Add closed? to MiniSSL::Socket for use in reactor ([#1510](https://github-redirect.dependabot.com/puma/puma/issues/1510))
>   * Handle EOFError at the toplevel of the server threads ([#1524](https://github-redirect.dependabot.com/puma/puma/issues/1524)) ([#1507](https://github-redirect.dependabot.com/puma/puma/issues/1507))
>   * Deal with zero sized bodies when using SSL ([#1483](https://github-redirect.dependabot.com/puma/puma/issues/1483))
> 
> ## 3.11.2 / 2018-01-19
> 
> * 1 bugfix:
>   * Deal with read\_nonblock returning nil early
> 
> ## 3.11.1 / 2018-01-18
> 
> * 1 bugfix:
>   * Handle read\_nonblock returning nil when the socket close ([#1502](https://github-redirect.dependabot.com/puma/puma/issues/1502))
> 
> ## 3.11.0 / 2017-11-20
> 
> * 2 features:
>   * HTTP 103 Early Hints ([#1403](https://github-redirect.dependabot.com/puma/puma/issues/1403))
>   * 421/451 status codes now have correct status messages attached ([#1435](https://github-redirect.dependabot.com/puma/puma/issues/1435))
> 
> * 9 bugfixes:
>   * Environment config files (/config/puma/<ENV>.rb) load correctly ([#1340](https://github-redirect.dependabot.com/puma/puma/issues/1340))
>   * Specify windows dependencies correctly ([#1434](https://github-redirect.dependabot.com/puma/puma/issues/1434), [#1436](https://github-redirect.dependabot.com/puma/puma/issues/1436))
>   * puma/events required in test helper ([#1418](https://github-redirect.dependabot.com/puma/puma/issues/1418))
>   * Correct control CLI's option help text ([#1416](https://github-redirect.dependabot.com/puma/puma/issues/1416))
>   * Remove a warning for unused variable in mini_ssl ([#1409](https://github-redirect.dependabot.com/puma/puma/issues/1409))
>   * Correct pumactl docs argument ordering ([#1427](https://github-redirect.dependabot.com/puma/puma/issues/1427))
>   * Fix an uninitialized variable warning in server.rb ([#1430](https://github-redirect.dependabot.com/puma/puma/issues/1430))
>   * Fix docs typo/error in Launcher init ([#1429](https://github-redirect.dependabot.com/puma/puma/issues/1429))
>   * Deal with leading spaces in RUBYOPT ([#1455](https://github-redirect.dependabot.com/puma/puma/issues/1455))
> 
> * 2 other:
>   * Add docs about internals ([#1425](https://github-redirect.dependabot.com/puma/puma/issues/1425), [#1452](https://github-redirect.dependabot.com/puma/puma/issues/1452))
>   * Tons of test fixes from [**MSP-Greg**](https://github.com/MSP-Greg) ([#1439](https://github-redirect.dependabot.com/puma/puma/issues/1439), [#1442](https://github-redirect.dependabot.com/puma/puma/issues/1442), [#1464](https://github-redirect.dependabot.com/puma/puma/issues/1464))
> 
> ## 3.10.0 / 2017-08-17
> 
> * 3 features:
>   * The status server has a new /gc and /gc-status command. ([#1384](https://github-redirect.dependabot.com/puma/puma/issues/1384))
>   * The persistent and first data timeouts are now configurable ([#1111](https://github-redirect.dependabot.com/puma/puma/issues/1111))
>   * Implemented RFC 2324 ([#1392](https://github-redirect.dependabot.com/puma/puma/issues/1392))
> 
> * 12 bugfixes:
>   * Not really a Puma bug, but [**NickolasVashchenko**](https://github.com/NickolasVashchenko) created a gem to workaround a Ruby bug that some users of Puma may be experiencing. See README for more. ([#1347](https://github-redirect.dependabot.com/puma/puma/issues/1347))
>   * Fix hangups with SSL and persistent connections. ([#1334](https://github-redirect.dependabot.com/puma/puma/issues/1334))
>   * Fix Rails double-binding to a port ([#1383](https://github-redirect.dependabot.com/puma/puma/issues/1383))
>   * Fix incorrect thread names ([#1368](https://github-redirect.dependabot.com/puma/puma/issues/1368))
> ... (truncated)
</details>
<details>
<summary>Commits</summary>

- [`4163dd7`](https://github.com/puma/puma/commit/4163dd71a11ed3b5b45799ed99c296e7df6ef3ba) Bump to 3.11.3
- [`61c875f`](https://github.com/puma/puma/commit/61c875f0de520076b77e43e9ffaca9547bd025ab) Deal with zero sized bodies properly when using SSL. Fixes [#1483](https://github-redirect.dependabot.com/puma/puma/issues/1483)
- [`b9ff900`](https://github.com/puma/puma/commit/b9ff900eee5e7b8a9d3eabcf69d4ac4fa0ffb4ef) Deal with early closes properly. Fixes [#1524](https://github-redirect.dependabot.com/puma/puma/issues/1524), [#1507](https://github-redirect.dependabot.com/puma/puma/issues/1507)
- [`1638b2a`](https://github.com/puma/puma/commit/1638b2af5aa5568a40898fe5244975c0df8ef7cd) Add closed to MinSSL. Fixes [#1510](https://github-redirect.dependabot.com/puma/puma/issues/1510)
- [`dc9fa77`](https://github.com/puma/puma/commit/dc9fa77f855c5018a03430366ea8c8db17fbfeea) Bump for 3.11.2
- [`49ed71a`](https://github.com/puma/puma/commit/49ed71ad3ea3e173a05de5c621c712549b496587) Deal with read_nonblock returning nil early
- [`beea8dc`](https://github.com/puma/puma/commit/beea8dc548ed76580753bbd5984779428e340009) Bump for 3.11.1
- [`0ed4ba8`](https://github.com/puma/puma/commit/0ed4ba847bc4d24a0bad92af1cc22b5bd2e9af12) Ruby 2.1 is now EOL
- [`acb709b`](https://github.com/puma/puma/commit/acb709b7b74ba7f8129d0fa7ffe4f04e6da24aa2) Bail if read_nonblock returns nil. Fixes [#1502](https://github-redirect.dependabot.com/puma/puma/issues/1502)
- [`9369a9f`](https://github.com/puma/puma/commit/9369a9fee0defba89a563d46a060ec31c4432c82) Ragel ver bump and regen
- Additional commits viewable in [compare view](https://github.com/puma/puma/compare/v3.9.1...v3.11.3)
</details>
<br />

Basically just bug fixes. I used the tool I built, [Dependabot](https://dependabot.com), to generate the PR and description above - would love to get you using it at Discourse.

### CODE REVIEW

**Comment by SamSaffron** [comment]
sure, seems fine to, thanks!

**Comment by SamSaffron** [comment]
The the frustrating thing about robot detecting stuff though is that they miss he human touch, unfortunately upgrading puma means that `bin/puma` in dev results in a very hard to kill puma that require me to run `ps aux` to hunt it down and kill -9 it. 

I am reverting this for now till someone with a dev env figures out what is up cc @nateberkopec
